examples/trossen_ai/main.py

```diff
@@ -0,0 +1,255 @@
+from lerobot.robots.bi_widowxai_follower.config_bi_widowxai_follower import BiWidowXAIFollowerConfig
+from lerobot.cameras.realsense.configuration_realsense import RealSenseCameraConfig
+from collections import defaultdict
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'lerobot'))
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+class TrossenOpenPIBridge:
+    """Bridge between a single Trossen arm and OpenPI policy server."""
+
+    def __init__(
+        self,
+        policy_server_host: str = "localhost",
+        policy_server_port: int = 8000,
+        control_frequency: int = 30,
+        test_mode: bool = False,
+    ):
+        self.control_frequency = control_frequency
+        self.dt = 1.0 / control_frequency
+        self.test_mode = test_mode
+
+        logger.info(f"Connecting to policy server at {policy_server_host}:{policy_server_port}")
+        self.policy_client = websocket_client_policy.WebsocketClientPolicy(
+            host=policy_server_host,
+            port=policy_server_port
+        )
+
+        bi_widowx_ai_config = BiWidowXAIFollowerConfig(
+            left_arm_ip_address="192.168.1.5",
+            right_arm_ip_address="192.168.1.4",
+            min_time_to_move_multiplier=4.0,
+            id="bimanual_follower",
+            cameras={
+                "top": RealSenseCameraConfig(
+                    serial_number_or_name="218622270304",
+                    width=640, height=480, fps=30, use_depth=False
+                ),
+                "bottom": RealSenseCameraConfig(
+                    serial_number_or_name="130322272628",
+                    width=640, height=480, fps=30, use_depth=False
+                ),
+                "right": RealSenseCameraConfig(
+                    serial_number_or_name="128422271347",
+                    width=640, height=480, fps=30, use_depth=False
+                ),
+                "left": RealSenseCameraConfig(
+                    serial_number_or_name="218622274938",
+                    width=640, height=480, fps=30, use_depth=False
+                ),
+            }
+        )
+        self.robot = make_robot_from_config(bi_widowx_ai_config)
+        self.robot.connect()
+
+        self.current_action_chunk = None
+        self.action_chunk_idx = 0
+        self.episode_step = 0
+        self.is_running = False
+        self.rate_of_inference = 50  # Number of control steps per policy inference
+
+        self.m = None  # Temporal ensembling weight (can be set to None for no ensembling)
+
+        # FIFO Buffer for actions
+        self.buffer = defaultdict(list)
+        self.action_buffer_size = 30 * 60  # Predefined buffer size (Depends on max episode length)
+
+    def execute_action(self, action: np.ndarray):
+        """Execute action on the arm."""
+        full_action = action.copy()
+        joint_limits = np.array([
```

Comment:
Some of these joint limits don't look right - make sure you're following the URDF or the specifications: https://docs.trossenrobotics.com/trossen_arm/main/specifications.html#joint-limits